File: 02Algorithm_Study/03_week_homeWork/Greedy_Che.py
def solution(n, lost, reserve):
    answer = 0
    che = [1] * (n)
    for i in reserve:
        che[i-1] += 1
    for i in lost:
        che[i-1] -= 1
    print("che",che)
    
    for i in range(n):
        for j in [-1,1]:
            ran = i + j
            if(ran<0 or ran >= n):
                continue;
            else:
                if(che[i] == 2 and che[ran] == 0):
                    che[i] -= 1
                    che[ran] += 1
    print("che",che)
    for i in che:
        if(i>0):
            answer += 1
            
    print("answer", answer)
    return answer

File: 02Algorithm_Study/03_week_homeWork/Greedy_Che.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int n, vector<int> lost, vector<int> reserve) {
    int answer = 0;
    vector <int> che;
    int tmp[2] = {-1,1};
    for(int i = 0; i < n ; i ++){
        che.push_back(1);
    }
    
    for (int i = 0; i < lost.size(); i ++){
        che[lost[i]-1] -=1;
    }
    for (int i = 0; i < reserve.size(); i++){
        che[reserve[i]-1] +=1;
    }
    
    for (int i = 0; i < n; i ++){
        for (int j = 0; j<2; j ++){
            int give = i + tmp[j];
            if(give<0 or give>=n){
                continue;
            }
            else{
                if(che[give] == 2 && che[i] == 0){
                    che[give] -=1;
                    che[i]+=1;
                }
            }
        }
    }
    for (int i = 0; i <n; i ++){
        //printf("%d ",che[i]);
        //cout << che[i] << ' ';
        if(che[i]>0){
        answer ++;
        }
    }
    return answer;
}
